langchain/vectorstores/weaviate.py

```diff
@@ -180,7 +189,39 @@ def max_marginal_relevance_search(
             payload[idx].pop("_additional")
             meta = payload[idx]
             docs.append(Document(page_content=text, metadata=meta))
-        return docs
+        return docs 
+
+    def similarity_search_with_score(
+        self, query: str, k: int = 4, **kwargs: Any
+    ) -> List[Tuple[Document, float]]:
+        content: Dict[str, Any] = {"concepts": [query]}
+        if kwargs.get("search_distance"):
+            content["certainty"] = kwargs.get("search_distance")
+        query_obj = self._client.query.get(self._index_name, self._query_attrs)
+        result = query_obj.with_near_text(content).with_limit(k).with_additional("vector").do()
+        if "errors" in result:
+            raise ValueError(f"Error during query: {result['errors']}")
+
+        docs_and_scores = []
+        for res in result["data"]["Get"][self._index_name]:
+            text = res.pop(self._text_key)
+            score = np.dot(res["_additional"]["vector"], self._embedding.embed_query(query))
```

Comment:
If the embedding function isn't normalized, this could result in a score thats outside the "relevance score" bounds (0, 1).  This is fine for the `similarity_search_with_score` method itself since the score is unitless / not interpreted. 

We could add a function to `langchain/math_utils`

```python
def scaled_cosine_similarity(X: Matrix, Y: Matrix) -> np.ndarray:
    """Cosine similarity scaled to the range [0, 1]."""
    return (cosine_similarity(X, Y) + 1) / 2
```

And then call here:
```
scaled_cosine_similarity([res["_additional"]["vector"]], [self._embedding.embed_query(query)])[0]
```